### PR TITLE
Cobranded footer

### PR DESCRIPTION
**Pull request by zopa-charlotte-walker**
Cobranding in footer 

NEW PR: https://github.com/ZopaPublic/react-components/pull/1146

### CODE REVIEW

File: src/components/molecules/ZopaFooter/ZopaFooter.tsx [hunk 1]
```diff
@@ -394,6 +394,8 @@ export interface FooterProps extends HTMLAttributes<HTMLDivElement> {
   mainCustomLegalCopy?: string | string[];
   additionalCopy?: string[];
   customFooterLinks?: CustomFooterLink[];
+  isCobranded?: boolean;
+  MainCobrandedLegalCopy?: any; //TODO: add type
```
Comment (zopa-charlotte-walker) [inline src/components/molecules/ZopaFooter/ZopaFooter.tsx:398]: struggling to type this

File: src/components/molecules/ZopaFooter/__snapshots__/ZopaFooter.test.tsx.snap [hunk 2]
```diff
@@ -219,6 +219,7 @@ exports[`<ZopaFooter /> renders the component with no a11y violations 1`] = `
   font-weight: 400;
   -webkit-text-decoration: none;
   text-decoration: none;
+  color: #3B46C4;
```
Comment (zopa-charlotte-walker) [inline src/components/molecules/ZopaFooter/__snapshots__/ZopaFooter.test.tsx.snap:222]: isn't a change, it looked like this before just wasn't specified (the links being purple on footer)

File: src/components/molecules/ZopaFooter/ZopaFooter.tsx (hunk 1 above)
Comment (admmasters) [reply src/components/molecules/ZopaFooter/ZopaFooter.tsx:398]: casing inconsistency

```suggestion
  mainCobrandedLegalCopy?: FooterProps['customFooterLinks']
```

I think is the type you want here

File: .changeset/flat-zebras-deliver.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@zopauk/react-components': minor
```
Comment (moshie) [inline .changeset/flat-zebras-deliver.md:2]: Patch :)

File: src/components/molecules/ZopaFooter/ZopaFooter.tsx [hunk 4]
```diff
@@ -402,9 +404,17 @@ const ZopaFooter = ({
   additionalCopy = [],
   mainCustomLegalCopy,
   customFooterLinks,
+  isCobranded,
+  MainCobrandedLegalCopy,
   ...rest
 }: FooterProps) => {
-  const theme = useThemeContext();
+  const themeContext = useThemeContext();
+  let theme;
```
Comment (moshie) [inline src/components/molecules/ZopaFooter/ZopaFooter.tsx:412]: Type this with whatever zopaTheme & themeContext are

**Review by moshie — CHANGES_REQUESTED** [review]

File: src/components/molecules/ZopaFooter/ZopaFooter.tsx (hunk 1 above)
Comment (zopa-charlotte-walker) [reply src/components/molecules/ZopaFooter/ZopaFooter.tsx:398]: `Type '() => React.JSX.Element' is not assignable to type 'CustomFooterLink[]'.`

**Review by admmasters — APPROVED** [review]

**Comment by zopa-charlotte-walker** [comment]
NEW PR: [#1146](https://github.com/ZopaPublic/react-components/pull/1146)

**Pull request closed without merging**